client/report.go

```diff
@@ -0,0 +1,95 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+)
+
+const reportEndpoint = "/transactions.xml"
+
+//ReportAppID - Report for the Application Id authentication pattern with serviceToken
+func (client *ThreeScaleClient) ReportAppID(appId string, token string, tokenType string, serviceId string, transactions ReportTransactions) (ApiResponse, error) {
+	values := parseQueries(transactions, url.Values{}, transactions.Usage, transactions.Log)
+
+	switch tokenType {
+	case "service_token":
+		values.Add("service_token", token)
+	case "provider_key":
+		values.Add("provider_key", token)
+	default:
+		return ApiResponse{}, errors.New("invalid token type value")
+	}
+
+	values.Add("app_id", appId)
+	values.Add("service_id", serviceId)
+
+	return client.report(values)
+}
+
+//ReportUserKey - Report for the API Key authentication pattern with service token
+func (client *ThreeScaleClient) ReportUserKey(userKey string, token string, tokenType string, serviceId string, transactions ReportTransactions) (ApiResponse, error) {
+	values := parseQueries(transactions, url.Values{}, transactions.Usage, transactions.Log)
```

Comment:
@jmprusi I think the `tokenType` is fine, but we may need to add it elsewhere. I'd also say we should expose two `consts` with their values to make it simpler to use.

Reply:
@PhilipGough Im planning on refactor all the other functions if we agree on this. 

so, provider_key and service_token to const.  ok!